version: "3.7"

volumes:
  database_data:
  parallax:

networks:
  default:

services:
  db:
    image: postgres:latest
    command: -p 5433
    volumes:
      - database_data:/var/lib/postgresql/data
    environment:
      - POSTGRES_USER=parallax
      - POSTGRES_DB=parallax-auth
      - POSTGRES_PASSWORD=parallax
      - POSTGRES_PORT=5433
    ports:
      - '5433:5433'
    networks:
      - default

  parallax-auth:
    build: .
    command: python manage.py runserver 0.0.0.0:8001
    volumes:
      - .:/parallax-auth
    expose:
      - 8001
    ports:
      - "8001:8001"
    environment:
      - PYTHONUNBUFFERED=1
      - DEBUG=True
      - DB_HOST=db
      - DB_NAME=parallax-auth
      - DB_USER=parallax
      - DB_PASSWORD=parallax
      - DB_PORT=5433
    links:
      - db
    depends_on:
      - db
    networks:
      - default

  tests:
    build: .
    command: coverage run -m pytest --cov -p no:cacheprovider
    volumes:
      - .:/parallax-auth
    environment:
      - PYTHONUNBUFFERED=1
      - DJANGO_SETTINGS_MODULE=parallax_auth.test_settings
      - DB_HOST=db
      - DB_NAME=parallax
      - DB_USER=parallax
      - DB_PASSWORD=parallax
    links:
      - db
    depends_on:
      - db
    networks:
      - default
